src/Management/src/EndpointCore/ManagementHostBuilderExtensions.cs

```diff
@@ -92,182 +86,195 @@ public static IHostBuilder AddHealthActuator(this IHostBuilder hostBuilder, Type
-        {
-            return hostBuilder
+            => hostBuilder
                 .ConfigureServices((context, collection) =>
                 {
                     collection.AddTraceActuator(context.Configuration, mediaTypeVersion);
-                    collection.AddTransient<IStartupFilter, TraceStartupFilter>();
+                    ActivateActuatorEndpoints(collection);
                 });
-        }
 
         /// <summary>
         /// Adds the Cloud Foundry actuator to the application
         /// </summary>
         /// <param name="hostBuilder">Your HostBuilder</param>
         public static IHostBuilder AddCloudFoundryActuator(this IHostBuilder hostBuilder)
-        {
-            return hostBuilder
+            => hostBuilder
                 .ConfigureServices((context, collection) =>
                 {
                     collection.AddCloudFoundryActuator(context.Configuration);
-                    collection.AddTransient<IStartupFilter, CloudFoundryActuatorStartupFilter>();
+                    ActivateActuatorEndpoints(collection);
                 });
-        }
 
+        /// <summary>
+        /// Adds all standard actuators to the application
+        /// </summary>
+        /// <param name="hostBuilder">Your HostBuilder</param>
+        /// <param name="configureEndpoints">Customize endpoint behavior. Useful for tailoring auth requirements</param>
+        /// <param name="mediaTypeVersion">Specify the media type version to use in the response</param>
+        /// <remarks>Does not add platform specific features (like for Cloud Foundry or Kubernetes)</remarks>
         public static IHostBuilder AddAllActuators(this IHostBuilder hostBuilder, Action<IEndpointConventionBuilder> configureEndpoints = null, MediaTypeVersion mediaTypeVersion = MediaTypeVersion.V2)
-        {
-            return hostBuilder
+            => hostBuilder
                 .AddDynamicLogging()
                 .ConfigureServices((context, collection) =>
                 {
                     collection.AddAllActuators(context.Configuration, mediaTypeVersion);
-                    collection.AddTransient<IStartupFilter, AllActuatorsStartupFilter>(provider => new AllActuatorsStartupFilter(configureEndpoints));
+                    ActivateActuatorEndpoints(collection, configureEndpoints);
                 });
+
+        /// <summary>
+        /// Registers an <see cref="IStartupFilter" /> that will map all configured actuators, initialize health
+        /// </summary>
+        /// <param name="collection"><see cref="IServiceCollection" /> that has actuators to activate</param>
+        /// <param name="configureEndpoints">IEndpointConventionBuilder customizations (such as auth policy customization)</param>
+        public static void ActivateActuatorEndpoints(this IServiceCollection collection, Action<IEndpointConventionBuilder> configureEndpoints = null)
```

Comment:
@hananiel this is new, please focus here

Reply:
Note: this method allows application of one unified set of endpoint convention customization for all actuator endpoints. If anybody wants to do per-endpoint customization they can/should use [`Map<TEndpoint>`](https://github.com/SteeltoeOSS/Steeltoe/blob/master/src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs#L62) directly for each endpoint instead of this more general one-size-fits-most option